auth/src/providers/VSCodeAzureSubscriptionProvider.ts

```diff
@@ -0,0 +1,191 @@
+        }
+
+        // Finally, sort
+        return results.sort((a, b) => {
+            if (a.displayName && b.displayName) {
+                return a.displayName.localeCompare(b.displayName);
+            }
+            return a.tenantId.localeCompare(b.tenantId);
+        });
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public override async getSubscriptionsForTenant(tenant: TenantIdAndAccount, options: GetSubscriptionsOptions = DefaultGetSubscriptionsOptions): Promise<AzureSubscription[]> {
+        const cacheKey = `${tenant.account.id.toLowerCase()}/${tenant.tenantId.toLowerCase()}`;
+
+        // If needed, delete the cache for this tenant
+        if (options.noCache) {
+            this.subscriptionCache.delete(cacheKey);
+        }
+
+        // If needed, refill the cache
+        if (!this.subscriptionCache.has(cacheKey)) {
+            const subscriptions = await super.getSubscriptionsForTenant(tenant, options);
+            this.subscriptionCache.set(cacheKey, subscriptions);
+            this.logger?.debug(`auth: Cached ${subscriptions.length} subscriptions for account '${tenant.account.id}' and tenant '${tenant.tenantId}'`);
+        }
+
+        let results: AzureSubscription[] = this.subscriptionCache.get(cacheKey)!; // eslint-disable-line @typescript-eslint/no-non-null-assertion -- We just filled it
+
+        // If needed, filter according to configured filters
+        if (!options.all) {
+            const subscriptionFilters = await this.getSubscriptionFilters();
+            if (subscriptionFilters.length > 0) {
+                this.logger?.debug(`auth: Filtering subscriptions for account '${tenant.account.id}' and tenant '${tenant.tenantId}' to ${subscriptionFilters.length} configured subscriptions`);
+                results = results.filter(sub => subscriptionFilters.includes(sub.subscriptionId.toLowerCase()));
+            }
+        }
+
+        // If needed, dedupe according to options
+        if (options.dedupe ?? true) {
+            this.logger?.debug(`auth: Deduping subscriptions for account '${tenant.account.id}' and tenant '${tenant.tenantId}'`);
+            results = dedupeSubscriptions(results);
+        }
+
+        // Finally, sort
+        return results.sort((a, b) => a.name.localeCompare(b.name));
+    }
+
+    /**
+     * Gets the account filters that are configured in `azureResourceGroups.selectedSubscriptions`. To
+     * override the settings with a custom filter, implement a child class with filter methods overridden.
+     *
+     * If no values are returned by `getAccountFilters()`, then all accounts will be scanned for subscriptions.
+     *
+     * @returns A list of account IDs that are configured in `azureResourceGroups.selectedSubscriptions`.
+     */
+    protected getAccountFilters(): Promise<string[]> {
+        // TODO: implement account filtering based on configuration if needed
+        // TODO: cannot have PII in the settings so it has to be an ID--test that
```

Comment:
I did, ID is ok. Still not sure how to impl this without using a new setting entirely